src/ovro_lwa_portal/io.py

```diff
@@ -0,0 +1,373 @@
+
+    Parameters
+    ----------
+    source : str or Path
+        Data source, can be:
+        - Local file path (e.g., "/path/to/data.zarr")
+        - Remote URL (e.g., "s3://bucket/data.zarr", "https://...")
+        - DOI string (e.g., "doi:10.xxxx/xxxxx" or "10.xxxx/xxxxx")
+    chunks : dict, str, or None, default "auto"
+        Chunking strategy for lazy loading:
+        - dict: Explicit chunk sizes per dimension, e.g., {"time": 100, "frequency": 50}
+        - "auto": Let xarray/dask determine optimal chunks
+        - None: Load entire dataset into memory (not recommended for large data)
+    engine : str, default "zarr"
+        Backend engine for loading data. Currently supports "zarr".
+    validate : bool, default True
+        If True, validate that loaded data conforms to OVRO-LWA data model.
+    **kwargs
+        Additional arguments passed to the underlying loader (xr.open_zarr, etc.)
+
+    Returns
+    -------
+    xr.Dataset
+        OVRO-LWA dataset with standardized structure.
+
+    Raises
+    ------
+    DataSourceError
+        If source cannot be accessed or loaded.
+    FileNotFoundError
+        If local file path doesn't exist.
+    ImportError
+        If required dependencies for remote/DOI access are not installed.
+
+    Examples
+    --------
+    Load from local zarr store:
+
+    >>> import ovro_lwa_portal
+    >>> ds = ovro_lwa_portal.open_dataset("/path/to/observation.zarr")
+
+    Load from S3 bucket:
+
+    >>> ds = ovro_lwa_portal.open_dataset("s3://ovro-lwa-data/obs_12345.zarr")
+
+    Load from HTTP/HTTPS URL:
+
+    >>> ds = ovro_lwa_portal.open_dataset("https://data.ovro.caltech.edu/obs_12345.zarr")
+
+    Load via DOI (with or without prefix):
+
+    >>> ds = ovro_lwa_portal.open_dataset("doi:10.5281/zenodo.1234567")
+    >>> ds = ovro_lwa_portal.open_dataset("10.5281/zenodo.1234567")
+
+    Customize chunking:
+
+    >>> ds = ovro_lwa_portal.open_dataset(
+    ...     "path/to/data.zarr",
+    ...     chunks={"time": 100, "frequency": 50}
+    ... )
+
+    Notes
+    -----
+    For remote data sources (S3, GCS), authentication is handled via environment
+    variables or configuration files specific to each cloud provider:
+
+    - AWS S3: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, or ~/.aws/credentials
+    - Google Cloud Storage: GOOGLE_APPLICATION_CREDENTIALS
+    - Azure: AZURE_STORAGE_ACCOUNT_NAME, AZURE_STORAGE_ACCOUNT_KEY
+
+    For large datasets, lazy loading with dask is used by default (chunks="auto").
+    This allows working with datasets larger than memory.
+    """
+    source_type, normalized_source = _detect_source_type(source)
+    logger.info(f"Detected source type: {source_type}")
+
+    # Resolve DOI to actual data URL
+    if source_type == "doi":
+        logger.info(f"Resolving DOI: {normalized_source}")
+        try:
+            normalized_source = _resolve_doi(normalized_source)
+            source_type = "remote"  # After resolution, treat as remote URL
```

Comment:
There's a lot of logic here that have potential to break. Let's use fsspec's universal pathlib for pathing so that it can handle either remote and local: https://github.com/fsspec/universal_pathlib

The input for `xr.open_zarr` should be a zarr store not just a path.